lib/dependency-manager-adapters/npm.js

```diff
@@ -7,13 +7,37 @@ var path = require('path');
 var extend = require('extend');
 var debug = require('debug')('ember-try:dependency-manager-adapter:npm');
 var rimraf = RSVP.denodeify(require('rimraf'));
+var execFileSync = require('child_process').execFileSync;
 
 module.exports = CoreObject.extend({
   init: function() {
     this._super.apply(this, arguments);
     this.run = this.run || require('../utils/run');
+    this.adapterInit();
   },
-  configKey: 'npm',
+  adapterInit() {
+    this.yarnLockPresent = fs.existsSync(path.join(this.cwd, this.yarnLock));
+    let yarnManagerPresent;
+    try {
+      execFileSync('yarn', [].concat(['--version']));
+      yarnManagerPresent = true;
+    } catch (e) {
+      yarnManagerPresent = false;
+    }
+    if (this.yarnLockPresent || yarnManagerPresent) {
```

Comment:
That implements the logic suggested by @kategengler 

> We have a bit of a different plan for yarn support: Rather than making a new adapter, yarn should be used in place of npm whenever yarn is installed **or** a yarn.lock is present. 

I also think that access to `yarn` binary should be required. Its clear that otherwise we might hit error otherwise. However i believe that we at least should notify the user, probably via `debug` message if `yarn.lock` file was present but we are not using it because we can't find the binary.